compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1151,29 +1151,35 @@ public void initPredeclaredModules(Map<Name, BPackageSymbol> predeclaredModules,
                                        List<BLangCompilationUnit> compUnits, SymbolEnv env) {
         SymbolEnv prevEnv = this.env;
         this.env = env;
-        for (Name alias : predeclaredModules.keySet()) {
+        Name alias;
+        for (Map.Entry<Name, BPackageSymbol> e : predeclaredModules.entrySet()) {
+            alias = e.getKey();
             int index = 0;
             ScopeEntry entry = this.env.scope.lookup(alias);
             if (entry == NOT_FOUND_ENTRY && !compUnits.isEmpty()) {
-                this.env.scope.define(alias, dupPackageSymbolAndSetCompUnit(predeclaredModules.get(alias),
+                this.env.scope.define(alias, dupPackageSymbolAndSetCompUnit(e.getValue(),
                         new Name(compUnits.get(index++).name)));
                 entry = this.env.scope.lookup(alias);
             }
+            if (!(entry.symbol instanceof BPackageSymbol)) {
```

Comment:
Here, We can use `SymbolKind` to do this check.

Reply:
The problem is kind is null here.

https://github.com/ballerina-platform/ballerina-lang/blob/b917d554c7da424ac3abb9a4cf86a1e784ab013f/compiler/ballerina-lang/src/main/java/io/ballerina/projects/Bootstrap.java#L167
This is the place where we set the package symbol, I think at this place we need to set the SymbolKind. 
@SasinduDilshara can you have a look?

Reply:
Please see https://github.com/ballerina-platform/ballerina-lang/pull/36459#discussion_r892001125. IMO, we first need to verify if the original logic is correct.

Reply:
Please check the this place too.
https://github.com/ballerina-platform/ballerina-lang/blob/4c95b7d97667a32d9187785a64dfb394820a3c32/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/symbols/Symbols.java#L58

At the first run, code goes through the 2nd condition and 2nd run code goes through 1st condition.

We may need to check whether we need to call `initPredeclaredModules` when the PackageID equals `ANNOTATIONS`. 
@chiranSachintha

Reply:
Yeah. The main reason for the above issue is that predeclared modules try to define within the annotation package(owner of the default package). In that instance, within the annotation package, it contains type symbols with the alias. (For example, the type symbol of the string contains with the `string` alias). So without defining predeclared modules use the type symbol within the `initPredeclaredModules` function and it will cause CCE.
Actually we didn't need to define predeclared symbols within the annotation module because the predeclared modules are already defined within the default module.